haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +438,95 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        boto3_import.check()
+
+        # See https://docs.aws.amazon.com/bedrock/latest/userguide/embeddings.html for more details
+        # The maximum input text is 8K tokens and the maximum output vector length is 1536
+        # Bedrock Titan embeddings do not support batch operations
+        self.aws_config = retriever.aws_config
+        self.client = self.initialize_boto3_session().client("bedrock-runtime")
+        self.model: str = next(
+            (m for m in BEDROCK_EMBEDDING_MODELS if m in retriever.embedding_model), "Model Not Found"
+        )
+
+        if self.model == "Model Not Found":
+            raise ValueError("Model not found in Bedrock Embedding Models")
+
+    def initialize_boto3_session(self):
+        if self.aws_config:
+            aws_profile_name = self.aws_config.get("profile_name", None)
+            aws_access_key_id = self.aws_config.get("aws_access_key_id", None)
+            aws_secret_access_key = self.aws_config.get("aws_secret_access_key", None)
+            aws_region_name = self.aws_config.get("region_name", None)
+            aws_session_token = self.aws_config.get("session_token", None)
+            try:
+                return boto3.Session(
+                    aws_access_key_id=aws_access_key_id,
+                    aws_secret_access_key=aws_secret_access_key,
+                    aws_session_token=aws_session_token,
+                    region_name=aws_region_name,
+                    profile_name=aws_profile_name,
+                )
+            except Exception as e:
+                raise ValueError(f"AWS client error {e}")
+
+    def embed(self, text: str) -> np.ndarray:
+        if self.model == "amazon.titan-embed-text-v1":
+            input_body = {}
+            input_body["inputText"] = text
+            body = json.dumps(input_body)
+            response = self.client.invoke_model(
+                body=body, modelId=self.model, accept="application/json", contentType="application/json"
+            )
+
+            response_body = json.loads(response.get("body").read())
+            return np.array(response_body.get("embedding"))
+        else:
+            coherePayload = json.dumps({"texts": [text], "input_type": "search_query", "truncate": "END"})
```

Comment:
I might have been a bit unclear on this:
- for embed_documents it must be `search_document`
- for embed_queries it must be `search_query`